### PR TITLE
docs: update documentation to use buf.gen.yaml v2

### PR DESCRIPTION
**Pull request by hirasawayuki**
The Buf CLI 1.32.0 introduced a v2 config format for buf.gen.yaml ([blog post](https://buf.build/blog/buf-cli-next-generation)). This PR updates the examples in documentation to use buf.gen.yaml v2 for generating code.

### CODE REVIEW

File: docs/go/getting-started.md
Comment (hirasawayuki) [inline docs/go/getting-started.md:92]: Command "init" is deprecated, use "buf config init" instead. However, "buf mod init" will continue to work.

File: docs/go/getting-started.md
Comment (hirasawayuki) [inline docs/go/getting-started.md:105]: exec `buf config migrate`

File: docs/node/getting-started.md
Comment (hirasawayuki) [inline docs/node/getting-started.md:48]: Specified that when the package is set to connectrpc.eliza.v1, the directory connectrpc/eliza/v1 must exist.

```
proto/eliza.proto:3:1:Files with package "connectrpc.eliza.v1" must be within a directory "connectrpc/eliza/v1" relative to root but were in directory "proto".
```

File: docs/node/getting-started.md
Comment (hirasawayuki) [inline docs/node/getting-started.md:81]: Updated the explanation for the Node.js example to align with those of other languages.

File: docs/go/getting-started.md
Comment (nicksnyder) [inline docs/go/getting-started.md:84]: I don't think it is necessary to reference the config version in the header because it is self evident in the config file itself. (if we did want to reference it in the header, the correct way would be to call it "buf.gen.yaml v2", but like I said, I don't think this is necessary for the header)

```suggestion
## Generate code
```

**Review by nicksnyder — COMMENTED** [review]
Appreciate the PR to update documentation to update buf.gen.yaml to the v2 format. The language in the PR title, PR description, and markdown headings is misleading though: there is no v2 of the Buf CLI. The most recent version of the Buf CLI is v1.32.0. Here is the blog post that explains more: https://buf.build/blog/buf-cli-next-generation

If you are referring to a v2 anywhere (in markdown, PR title, PR description, etc), you should specify that it is the configs you are talking about.

File: docs/go/getting-started.md
Comment (hirasawayuki) [reply docs/go/getting-started.md]: Absolutely. I have fixed it with [a2b3f3f](https://github.com/connectrpc/connectrpc.com/pull/162/commits/a2b3f3fd68f9107ceb7ac17f89e5cdddf564859e)

File: docs/kotlin/getting-started.md
Comment (pkwarren) [inline docs/kotlin/getting-started.md:65]: We need to update above on line 62 as well to replace `mod init` with `config init`.

File: docs/kotlin/getting-started.md
Comment (pkwarren) [inline docs/kotlin/getting-started.md:114]: Our recommendation is to put both `buf.yaml` and `buf.gen.yaml` at the root of the repository going forward, so I think these instructions need to be rewritten as:

```
$ buf config init
```

Then edit `buf.yaml` to add:
```yaml
modules:
  - path: proto
```

Then run:
```
$ buf export buf.build/connectrpc/eliza -o proto
```

Then we'd remove the `cd ..` step and the `buf generate proto` command can become just `buf generate`.

File: docs/node/getting-started.md
Comment (pkwarren) [inline docs/node/getting-started.md:75]:
```suggestion
configuration files to get going. (If you'd prefer, you can skip this section
```

File: docs/swift/getting-started.md
Comment (pkwarren) [inline docs/swift/getting-started.md:92]: This looks out of place - to change the `eliza.proto` steps you need to update the section above.

**Comment by hirasawayuki** [comment]
@pkwarren 
Thank you for your review! Based on your feedback, I've made revisions in commit 43c78d7.

Unified the documentation for using ElizaService across Node, Kotlin, and Swift as much as possible.
Could you please review it again?

**Comment by pkwarren** [comment]
> Thank you for your review! Based on your feedback, I've made revisions in commit 43c78d7.

Thank you so much for making the updates. I will look first thing on Tuesday when we get back from a holiday weekend.

If there are small tweaks to the docs are you ok with me making edits in this PR?

**Comment by hirasawayuki** [comment]
> If there are small tweaks to the docs are you ok with me making edits in this PR?

Sure, feel free to make any edits. Thank you!

File: docs/node/getting-started.md
Comment (timostamm) [inline docs/node/getting-started.md:44]: With Connect for Node.js, users can serve RPCs. We're showing how to do that in this tutorial. It's most common that users who implement a service also author the schema. In this context, I'd prefer that we let the user author greet.proto, instead of exporting it from the BSR. Similar to what we do in the connect-go docs.

File: docs/node/getting-started.md
Comment (timostamm) [inline docs/node/getting-started.md:57]: The next page "Implementing services" builds on this page, and imports from `"./gen/eliza_connect"`. If we choose to use the directory tree (I'm not opposed to it), these imports have to be updated as well. There may be more pages that need to be updated.

**Review by timostamm — COMMENTED** [review]
Thank you for the PR, @hirasawayuki 🙏 

I noticed we need some more updates, see comments below, and the pages "Generating code" and "Supported browsers & frameworks" under "Connect for Web" - they use v1 buf.gen.yaml files.

File: docs/web/supported-browsers-and-frameworks.mdx
Comment (hirasawayuki) [reply docs/web/supported-browsers-and-frameworks.mdx]: I have made the updates in commit [2152108](https://github.com/connectrpc/connectrpc.com/pull/162/commits/21521085b705a606c28880d9ac7efe8b28df39f7) !

**Comment by nicksnyder** [comment]
Hi @hirasawayuki, there are some other changes we want to make to this PR before merging, but it is going to be more efficient if you allow us to push directly rather than going back and forth on PR comments. Can you configure your fork to allow edits from maintainers?

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests

File: docs/go/getting-started.md
Comment (pkwarren) [inline docs/go/getting-started.md:99]:
```suggestion
  - local: protoc-gen-go
```

**Review by pkwarren — APPROVED** [review]

**Pull request merged**
